content/blog/reflections-of-a-pulumi-intern/index.md

```diff
@@ -0,0 +1,66 @@
+
+Over the course of the summer, I progessively built `crd2pulumi`, a CLI tool that generates a typed CustomResource from a CRD. You can find the latest releases [here](https://github.com/pulumi/crd2pulumi). I then leveraged this tool to generate libraries for various Kubernetes Operators, which you can find [here](https://github.com/pulumi/pulumi-kubernetes-crds/). I still feel like there's additional features to build out in this area though, so I definitely plan to submit a PR here and there even after today.
+
+## School vs. Work
+
+Prior to joining Pulumi, all my programming experiences had come from either school projects or practicing coding challenge type problems. So far, these scenarios have been fairly [atomic](https://wiki.osdev.org/Atomic_operation), in the sense that they are self-contained and started and finished in a relatively discrete chunk of time. A typical coding project at school is straightfoward—I sit down, read the instructions, code everything, and don't think about it until my grade comes back. Any problem can usually be resolved by looking more closely at the instructions or consulting the Piazza board.
+
+In contrast, real software development at Pulumi was far more multi-person, rich, and dynamic. It was annoying at times, but also far more rewarding. In the first couple of weeks, I was seemingly running into issues at all layers—with the `pulumi/pulumi` codegen, with the Go compiler, with the Pulumi CLI, with the Python SDKs, or with my Macbook overheating and crashing whenever I tried opening up the `examples` repository in VSCode. Getting stuff to "just work" often required fixing many nested sub-problems until the root problem could finally be resolved. After a while, I internalized a lot of general debugging heurstics, and developed better skills in squashing bugs and getting code to "just work."
+
+I thought that Pulumi gave me an excellent balance between guidance and autonomy. My co-workers gave me an outline on what my project should acheive, and some pointers on how which tools to use, but other than that it was up to me to implement it.
+
+One aspect that I liked about Pulumi was that its main audience is of course developers, so it was easier to design `crd2pulumi` since I myself am a developer. Being able to see and directly interact with other programmers that use my tool and offer feedback was immensely satisfying. I loved the energy that came from working with a fast-growing startup that provided real value to real people. It was exciting to come to Monday's team meeting and see Pulumi's user count contiously rise, and hear about the latest deal that the sales team had just closed.
```

Comment:
Typo: contiously => continuously

```suggestion
One aspect that I liked about Pulumi was that its main audience is of course developers, so it was easier to design `crd2pulumi` since I myself am a developer. Being able to see and directly interact with other programmers that use my tool and offer feedback was immensely satisfying. I loved the energy that came from working with a fast-growing startup that provided real value to real people. It was exciting to come to Monday's team meeting and see Pulumi's user count continuously rise, and hear about the latest deal that the sales team had just closed.
```